src/api/resolveRef.js

```diff
@@ -34,10 +35,11 @@ export async function resolveRef({
     assertParameter('fs', fs)
     assertParameter('gitdir', gitdir)
     assertParameter('ref', ref)
+    const updatedgitdir = await discoverGitdir(fs, dir, gitdir)
 
     const oid = await GitRefManager.resolve({
       fs: new FileSystem(fs),
-      gitdir,
```

Comment:
`new FileSystem(fs)` is called twice for no reason.

Reply:
"new FileSystem(fs) twice"  
Should be fixed.